### PR TITLE
Adds MapStoreConfig equals & hashCode implementations

### LINKED ISSUE #7035: MapStoreConfig does not override hashCode / equals methods
as a result this code is failing:

```java
MapConfig config1 = new MapConfig();
MapConfig config2 = new MapConfig();
assertEquals(config1, config2);
```

### PR DESCRIPTION
**Pull request by vbekiaris**
Fixes #7035

### CODE REVIEW

**Comment by jerrinot** [comment]
:+1:

**Comment by Donnerbart** [comment]
Just saying: This might be a little nightmare for the code coverage. Those `equals()`/`hashcode()` methods are always very tedious to test (line coverage might be okay with the single return line, but branch coverage might be hard to achieve). If this functionality is just needed for tests, I tend to create customized assert methods in the test and avoid `equals()`/`hashcode()` methods in the production code.

Beside that the PR looks fine to me :+1:

**Comment by vbekiaris** [comment]
Thanks @jerrinot, @Donnerbart . The new commit improves test coverage and makes an API change from `public void setWriteCoalescing(boolean writeCoalescing)` to `public MapStoreConfig setWriteCoalescing(boolean writeCoalescing)`; this allows chaining (as other setters in `MapStoreConfig` already do) and should not break any existing usages.

File: hazelcast/src/test/java/com/hazelcast/config/MapStoreConfigTest.java [hunk 1]
```diff
@@ -12,24 +12,45 @@
...
 import static com.hazelcast.config.MapStoreConfig.InitialLoadMode.EAGER;
 import static com.hazelcast.config.MapStoreConfig.InitialLoadMode.LAZY;
 import static org.junit.Assert.*;
 
 /**
  * Test MapStoreConfig
  */
-@RunWith(HazelcastParallelClassRunner.class)
-@Category({QuickTest.class, ParallelTest.class})
+//@RunWith(HazelcastParallelClassRunner.class)
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/config/MapStoreConfigTest.java]: I think these two lines should be enabled again, so the test is picked up by the correct Jenkins build.

**Comment by Donnerbart** [comment]
(forget the comment I just made, it was on an outdated diff - I used this "view only new changes" and it seemed to fool me)

**Comment by vbekiaris** [comment]
Actually you were right, they were commented out and I restored them; it looks like IDEA coverage with tracing interacts badly with the parallel test runner and random tests fail with an NPE:
```
java.lang.NullPointerException
	at com.intellij.rt.coverage.data.ProjectData.traceLine(ProjectData.java:349)
	at com.intellij.rt.coverage.data.ProjectData.trace(ProjectData.java:287)
	at com.hazelcast.config.MapStoreConfig.<init>(MapStoreConfig.java:70)
	at com.hazelcast.config.MapStoreConfigTest.<init>(MapStoreConfigTest.java:29)
	at sun.reflect.GeneratedConstructorAccessor4.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at org.junit.runners.BlockJUnit4ClassRunner.createTest(BlockJUnit4ClassRunner.java:217)
	at com.hazelcast.test.AbstractParameterizedHazelcastClassRunner.createTest(AbstractParameterizedHazelcastClassRunner.java:88)
	at org.junit.runners.BlockJUnit4ClassRunner$1.runReflectiveCall(BlockJUnit4ClassRunner.java:266)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.BlockJUnit4ClassRunner.methodBlock(BlockJUnit4ClassRunner.java:263)
	at com.hazelcast.test.AbstractHazelcastClassRunner.methodBlock(AbstractHazelcastClassRunner.java:168)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at com.hazelcast.test.HazelcastParallelClassRunner.access$301(HazelcastParallelClassRunner.java:46)
	at com.hazelcast.test.HazelcastParallelClassRunner$TestRunner.run(HazelcastParallelClassRunner.java:139)
	at java.lang.Thread.run(Thread.java:744)
```

**Comment by Donnerbart** [comment]
I've seen that kind of NPE before, but never made a connection to the parallel test runner. Good to know, thanks for pointing that out!

**Pull request merged**